src/app/new-auth/login/login.component.ts

```diff
@@ -1,23 +1,149 @@
-import { Component, OnInit } from '@angular/core';
+import { animate, style, transition, trigger } from '@angular/animations';
+import { Component, OnInit, ViewChild } from '@angular/core';
 import { NgForm } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { environment } from '@env/environment';
 import { AuthValidationService } from 'app/core/auth-validation.service';
+import { AuthService } from 'app/core/auth.service';
 
 @Component({
   selector: 'clark-login',
   templateUrl: './login.component.html',
-  styleUrls: ['./login.component.scss']
+  styleUrls: ['./login.component.scss'],
+  animations: [
+    trigger(
+      'inOutAnimation',
+      [
+        transition(
+          ':enter',
+          [
+            style({ height: 0, opacity: 0 }),
+            animate('.5s ease-out',
+                    style({ height: 85, opacity: 1 }))
+          ]
+        ),
+        transition(
+              ':leave',
+              [
+                style({ height: 85, opacity: 1 }),
+                animate('.5s ease-in',
+                        style({ height: 0, opacity: 0 }))
+              ]
+            )
+          ]
+        )
+      ]
 })
 export class LoginComponent implements OnInit{
 
-  loginFailure = false;
-  constructor(public authValidation: AuthValidationService) { }
+  finalAttempt: number;
+  loginFailure: Boolean = false;
+  isNameLogin = false;
+  authInfo: {username: string, password: string};
+  errorMsg = '';
+  bannerMsg = '';
+  attempts = 0;
+  @ViewChild('username')
+  username;
+  @ViewChild('password')
+  password;
+  redirectUrl;
+  gatewayUrl = environment.apiURL;
```

Comment:
For better readability:
```suggestion
  finalAttempt: number;
  errorMsg = '';
  bannerMsg = '';
  attempts = 0;
  loginFailure: Boolean = false;
  isNameLogin = false;
  
  authInfo: {
      username: string, 
      password: string
  };
  
  @ViewChild('username')
  username;
  @ViewChild('password')
  password;
  
  redirectUrl;
  gatewayUrl = environment.apiURL;
```